src/blockchains/eth/lib/decode_transfers.ts

```diff
@@ -1,8 +1,16 @@
-const { logger } = require('../../../lib/logger');
+import { Web3Interface } from './web3_wrapper';
+import { Trace, ETHTransfer } from '../eth_types';
+import { logger } from '../../../lib/logger';
 
-function decodeTransferTrace(trace, timestamp, web3Wrapper) {
+export function decodeTransferTrace(trace: Trace, timestamp: number, web3Wrapper: Web3Interface): ETHTransfer {
   // Block & uncle rewards
   if (trace['type'] === 'reward') {
+    if (trace['action']['author'] === undefined) {
```

Comment:
this pattern looks like assertion and TS has rich assertion capabilities
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions